src/lib/components/controls/ControlsList.svelte

```diff
@@ -119,15 +119,83 @@
 				results = results.filter((control) => {
 					// Control must match all filters
 					return $activeFilters.every(filter => {
+						// Handle mapping-related fields specially
+						if (filter.fieldName === 'has_mappings') {
+							const hasMappings = control.mappings && control.mappings.length > 0;
+							const filterValue = getFilterValue(filter.value);
+							switch (filter.operator) {
+								case 'equals':
+									return hasMappings === (filterValue === 'true' || filterValue === true);
+								case 'not_equals':
+									return hasMappings !== (filterValue === 'true' || filterValue === true);
+								case 'exists':
+									return hasMappings;
+								case 'not_exists':
+									return !hasMappings;
+								default:
+									return true;
+							}
+						}
+
+						if (filter.fieldName === 'mapping_count') {
+							const mappingCount = control.mappings ? control.mappings.length : 0;
+							const filterValue = getFilterValue(filter.value);
+							switch (filter.operator) {
+								case 'equals':
+									return mappingCount === Number(filterValue);
+								case 'not_equals':
+									return mappingCount !== Number(filterValue);
+								case 'exists':
+									return mappingCount > 0;
+								case 'not_exists':
+									return mappingCount === 0;
+								default:
+									return true;
+							}
+						}
```

Comment:
Maybe fine to just add it, but wondering if there's value in mapping count filters? Feel like we mostly care about 0 vs 1+ (which the `has mappings` would handle).

If we do want to keep count filters I might lean towards changing the operators to make more sense for numerical values (equals, not equals, great than, less than - rather than contains/not contains).